native/lambda_game_engine/src/game.rs

```diff
@@ -338,10 +347,15 @@ fn get_next_id(next_id: &mut u64) -> u64 {
 fn collect_nearby_loot(loots: &mut Vec<Loot>, player: &mut Player) {
     loots.retain(|loot| {
         if map::hit_boxes_collide(&loot.position, &player.position, loot.size, player.size) {
-            loot.effects
-                .iter()
-                .for_each(|effect| player.apply_effect(effect, EntityOwner::Loot));
-            false
+            match loot.pickup_mechanic {
+                PickupMechanic::CollisionToInventory => player.put_in_inventory(loot),
+                PickupMechanic::CollisionUse => {
+                    loot.effects
+                        .iter()
+                        .for_each(|effect| player.apply_effect(effect, EntityOwner::Loot));
```

Comment:
there's an existing function named `apply_effects` in `player.rs` that does this same thing. Can we re-use it? (i can't add the proper indentation here)
```suggestion
                    player.apply_effects(&loot.effects, EntityOwner::Loot);
```

Reply:
Missed that when moving this, fixed in [83220fb](https://github.com/lambdaclass/game_backend/pull/38/commits/83220fbf56cb7ca2365e3180f75a915afe0c858c)